hpux/generate_wazuh_packages.sh

```diff
@@ -115,80 +103,151 @@ check_version(){
 compile() {
   echo "Compiling code"
   # Compile and install wazuh
-  cd $SOURCE/src
+  cd ${source_directory}/src
   config
   check_version
   if [ "$deps_version" = "true" ]; then
     gmake deps RESOURCES_URL=http://packages.wazuh.com/deps/$major.$minor
   fi
 
   gmake TARGET=agent USE_SELINUX=no DISABLE_SHARED=yes
-  bash $SOURCE/install.sh
+  bash ${source_directory}/install.sh
   cd $current_path
 }
 
 create_package() {
   echo "Creating package"
+
+  if [ ! -d ${target_dir} ]; then
+    mkdir -p ${target_dir}
+  fi
+
   #Build package
   VERSION=`cat /tmp/VERSION`
-  rm $INSTALL/wodles/oscap/content/*.xml
-  tar cvpf /tmp/wazuh-agent-$VERSION-$REVISION-hpux-11v3-ia64.tar $INSTALL /etc/ossec-init.conf /sbin/init.d/wazuh-agent /sbin/rc2.d/S97wazuh-agent /sbin/rc3.d/S97wazuh-agent
+  rm ${install_path}/wodles/oscap/content/*.xml
+  pkg_name="wazuh-agent-${wazuh_version}-${wazuh_revision}-hpux-11v3-ia64.tar"
+  tar cvpf ${target_dir}/${pkg_name} ${install_path} /etc/ossec-init.conf /sbin/init.d/wazuh-agent /sbin/rc2.d/S97wazuh-agent /sbin/rc3.d/S97wazuh-agent
+
+  if [ "${compute_checksums}" = "yes" ]; then
+    cd ${target_dir}
+    pkg_checksum="$(openssl dgst -sha512 ${pkg_name})"
+    echo "${pkg_checksum}  ${pkg_name}" > ${checksum_dir}/${pkg_name}.sha512
+  fi
 }
 
 #Uninstall agent.
 
 clean() {
-  $INSTALL/bin/ossec-control stop
-  rm -rf $INSTALL
+  exit_code=$1
+  ${install_path}/bin/ossec-control stop
+  rm -rf ${install_path}
   rm /etc/ossec-init.conf
   find /sbin -name "*wazuh-agent*" -exec rm {} \;
   userdel ossec
   groupdel ossec
+
+  exit ${exit_code}
 }
 
-show_help()
-{
-  echo "
-  This scripts build wazuh package for HPUX.
-  USAGE: Command line options available:
-    -h, --help       Displays this help.
-    -d, --download   Download Wazuh repository.
-    -b, --build      Builds HPUX package.
-    -u, --utils      Download and install utilities and dependencies.
-    -c, --clean-all  Clean sources and generated files.
-
-  USAGE EXAMPLE:
-  --------------
-    ./generate_wazuh_packages.sh [option] [branch_tag] [revision]
-    ./generate_wazuh_packages.sh -d branches/3.3 1
-  "
+show_help() {
+  echo
+  echo "Usage: $0 [OPTIONS]"
+  echo
+  echo "    -e Install all the packages necessaries to build the TAR package"
+  echo "    -b <branch> Select Git branch. Example v3.5.0"
+  echo "    -s <tar_directory> Directory to store the resulting tar package. By default: /tmp/build"
```

Comment:
The default path has been changed right?

Reply:
Fixed here: https://github.com/wazuh/wazuh-packages/commit/46186f6714044a0b3d5447d968376bbc54a0f6fe